apps/prairielearn/src/pages/instructorAssessmentAccessOverrides/instructorAssessmentAccessOverrides.js

```diff
@@ -0,0 +1,196 @@
+    }
+    const group_result = await sqldb.queryZeroOrOneRowAsync(sql.select_group_in_assessment, {
+      group_name,
+      course_instance_id,
+      assessment_id: assessment.id,
+    });
+    if (group_result.rows.length > 0) {
+      return { user_id: null, group_id: group_result.rows[0].id };
+    } else {
+      throw error.make(400, 'Group not found in this assessment.');
+    }
+  }
+  if (uid) {
+    if (group_name) {
+      throw error.make(400, 'Student UID is required for individual work assessments.');
+    }
+    const user = await selectUserEnrolledInCourseInstance({
+      uid,
+      course_instance_id,
+    });
+    if (!user) {
+      throw error.make(400, `User ${uid} is not enrolled in this course instance.`);
+    }
+
+    return { user_id: user.user_id, group_id: null };
+  } else {
+    throw error.make(400, 'Student UID or Group Name is required.');
+  }
+}
+
+async function selectUserEnrolledInCourseInstance({ uid, course_instance_id }) {
+  const user = await selectUserByUid(uid);
+  if (!user) return null;
+
+  const enrollment = await getEnrollmentForUserInCourseInstance({
+    user_id: user.user_id,
+    course_instance_id,
+  });
+  if (!enrollment) return null;
+
+  return user;
+}
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_view) {
+      throw error.make(403, 'Access denied (must be a student data viewer)');
+    }
+    if (!res.locals.assessment_access_overrides_enabled) {
+      throw error.make(403, 'Access denied (feature not available)');
+    }
+
+    const result = await sqldb.queryAsync(sql.select_assessment_access_policies, {
+      assessment_id: res.locals.assessment.id,
+      timezone: res.locals.course_instance.display_timezone,
+    });
+    res.locals.policies = result.rows;
+    res.locals.timezone = res.locals.course_instance.display_timezone;
+    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_edit) {
+      throw error.make(403, 'Access denied (must be a student data editor)');
+    }
+    if (!res.locals.assessment_access_overrides_enabled) {
+      throw error.make(403, 'Access denied (feature not available)');
+    }
+
+    if (req.body.__action === 'add_new_override') {
+      const { user_id, group_id } = await getUserOrGroupId({
+        course_instance_id: res.locals.course_instance.id,
+        assessment: res.locals.assessment,
+        uid: req.body.student_uid,
+        group_name: req.body.group_name,
+      });
+      await runInTransactionAsync(async () => {
+        const inserted = await sqldb.queryOneRowAsync(sql.insert_assessment_access_policy, {
+          assessment_id: res.locals.assessment.id,
+          created_by: res.locals.authn_user.user_id,
+          credit: req.body.credit,
+          end_date: new Date(req.body.end_date),
```

Comment:
I worry slightly about parsing this as a `Date`. Is this necessary given that in SQL we have `$end_date::timestamp AT TIME ZONE $timezone` and such?

Reply:
No, and I struggled with formatting correctly in the rows when converting server side so this is working in safari and chrome and is correctly handling timezones. Let me know what you think 
355169b